contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -727,7 +727,12 @@ The only way for a client to know the expected value of resourceVersion is to
 have received it from the server in response to a prior operation, typically a
 GET. This value MUST be treated as opaque by clients and passed unmodified back
 to the server. Clients should not assume that the resource version has meaning
-across namespaces, different kinds of resources, or different servers.
+across different kinds of resources or different servers. Clients should not
+assume that the resource version of an object has meaning with respect to other
+objects. The resource version of a list on the other hand, represents the common
+version of the objects in the list results; however, clients that process the
+objects in the list results one at a time should only consider the individual
+objects' resource versions and ignore the resource version of the list.
```

Comment:
- I find "common version of the objects in the list results" vague. There are multiple ways to define "common"...
- I'm not sure the latter "should" clause for clients processing "one at a time" is a good contract. A developer can always say "ok, I'm considering whole list" and be "allowed" to use the list version? What is this guidance trying to encourage/prevent?

My 2¢ (as API consumer, I don't know the internals):
I've always thought that collection's version means I'm seeing a consistent snapshot at that "moment" ¹.
Individual object versions will likely be older — it's the last when that specific object changed, up to that moment — but if I list again providing the ListMeta.resourceVersion I got, I'll see exact same list (or 410 Gone), and if I list+watch using ListMeta.resourceVersion, I'll not miss anything (or get 410).

¹ Is it valid to imagine a whole collection evolving on a single linear timeline? Or does that break under network partitions etc.?